test/sdf_test.cpp

```diff
@@ -53,6 +61,162 @@ TEST(SDF, CubeVoid) {
   EXPECT_EQ(voidSDF({-2, 2, 2}), 1);
 }
 
+TEST(SDF, BatchValidation) {
+  const float size = glm::two_pi<float>();
+  const float edgeLength = size / 15;
+
+  Mesh levelSet = LevelSet(UnboundedGyroidSDF(),
+                           {glm::vec3(-size), glm::vec3(size)}, edgeLength);
+  Mesh levelSetBatch = LevelSetBatch(
+      [](std::vector<glm::vec3> pvec) {
+        std::vector<float> dvec(pvec.size());
+        for (int i = 0; i < pvec.size(); i++) {
+          dvec[i] = UnboundedGyroidSDF()(pvec[i]);
+        }
+        return dvec;
+      },
+      {glm::vec3(-size), glm::vec3(size)}, edgeLength);
+
+  // EXPECT_EQ(levelSet.vertPos .size(), levelSetBatch.vertPos .size());
```

Comment:
Not working?

Reply:
The triangulation is very slightly different between the batched and non-batched versions.

I suspect this is because of this check I had to add to avoid sampling out-of-bounds queries, which are hard to see ahead of time:
```cpp
        if (neighborCode >= gridDistances.size())
          continue;  // Neighbors outside of the bounds...
```
In practice, a mesh from the original function might have `95,078` triangles, while the batched function's might have `95,094` triangles.  It seems to fluctuate between having more and less than the original.

Reply:
That's a little troubling - I think of batch as being purely a performance optimization; I'm surprised it can come up with a different number of triangles or verts. Can you make a tiny example so we can see where the difference is? Do our tests at least verify that both versions are manifold and have the same volume and surface area?